ddtrace/llmobs/_integrations/utils.py

```diff
@@ -316,26 +409,28 @@ def openai_set_meta_tags_from_chat(
         processed_message = {
             "content": str(_get_attr(m, "content", "")),
             "role": str(_get_attr(m, "role", "")),
-        }  # type: dict[str, Union[str, list[dict[str, dict]]]]
+        }  # type: dict[str, Union[str, List[ToolCall], List[ToolResult]]]
         tool_call_id = _get_attr(m, "tool_call_id", None)
         if tool_call_id:
             core.dispatch(DISPATCH_ON_TOOL_CALL_OUTPUT_USED, (tool_call_id, span))
-            processed_message["tool_id"] = tool_call_id
-        tool_calls = _get_attr(m, "tool_calls", [])
-        if tool_calls:
-            processed_message["tool_calls"] = [
-                {
-                    "name": _get_attr(_get_attr(tool_call, "function", {}), "name", ""),
-                    "arguments": _get_attr(_get_attr(tool_call, "function", {}), "arguments", {}),
-                    "tool_id": _get_attr(tool_call, "id", ""),
-                    "type": _get_attr(tool_call, "type", ""),
-                }
-                for tool_call in tool_calls
-            ]
+
+        extracted_tool_calls, extracted_tool_results = _openai_extract_tool_calls_and_results_chat(m)
+
+        if extracted_tool_calls:
+            processed_message["tool_calls"] = extracted_tool_calls
+        if extracted_tool_results:
+            processed_message["tool_results"] = extracted_tool_results
+            processed_message["content"] = ""  # set content to empty string to avoid duplication
         input_messages.append(processed_message)
     parameters = get_metadata_from_kwargs(kwargs, integration_name, "chat")
     span._set_ctx_items({INPUT_MESSAGES: input_messages, METADATA: parameters})
 
+    if kwargs.get("tools") or kwargs.get("functions"):
+        tools = _openai_get_tool_definitions(kwargs.get("tools", []) or [])
+        tools.extend(_openai_get_tool_definitions(kwargs.get("functions", []) or []))
```

Comment:
```suggestion
        tools.extend(_openai_get_tool_definitions(kwargs.get("functions") or []))
```
No need to default to a value if we're going to default to the same value outside